src/js/_enqueues/lib/ajax-response.js

```diff
@@ -11,7 +11,7 @@ window.wpAjax = jQuery.extend( {
 		q = s.split('?'); if ( q[1] ) { s = q[1]; }
 		pp = s.split('&');
 		for ( i in pp ) {
-			if ( jQuery.isFunction(pp.hasOwnProperty) && !pp.hasOwnProperty(i) ) { continue; }
+			if ( typeof pp.hasOwnProperty === 'function' && !pp.hasOwnProperty(i) ) { continue; }
```

Comment:
Nit: Yoda

🔢 (applies in multiple places)

Reply:
Are the JavaScript conditionals when using type-checking meant to be Yoda'ed? (I'm not sure since the linter didn't complain about this, so just wanting to confirm)

Reply:
I too am used to using Yoda syntax everywhere, but after further examination I see it's in the [PHP Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions) but not a part of the [JavaScript Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/) so I suppose it's OK to use these as-is :thinking:

Reply:
Here, Yoda does not seem to apply. Why? Yoda seeks to prevent an assignment typo, i.e. `=` instead of `===`. However, in this case, such a typo will result in a `SyntaxError: Invalid left-hand side in assignment`. 

Why? The `typeof` operator will return a string for the type of given object or primitive. Then that string is used for the next operation. If `===` was an assignment typo, i.e. `=`, the resultant would be a string attempting to assign to a string (not a variable)

```
'function' = 'function'
```

which then throws the `SyntaxError`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

We can see how Gutenberg puts `typeof` on the left and the comparison on the right https://github.com/WordPress/gutenberg/search?q=typeof.

I'd advocate that Yoda does not apply here and will make the code harder to quickly understand.

Reply:
> I too am used to using Yoda syntax everywhere, but after further examination I see it's in the PHP Coding Standards but not a part of the JavaScript Coding Standards so I suppose it's OK to use these as-is 🤔

For context, noting that our JavaScript standards did originally call for Yoda conditionals to match the PHP spec as closely as possible. We changed that along with several other rules after discussion on the JavaScript team, see https://make.wordpress.org/core/2018/12/18/proposed-revisions-to-javascript-coding-standards/. 

We have also been discussing removing PHP yoda conditions entirely in https://github.com/WordPress/WordPress-Coding-Standards/issues/1624